### PR TITLE
Feature improve context for parallel execution

### PR DESCRIPTION
**Pull request by gkoch78**
so that visible messages in the context for a skill are limited to the parallel branch it is part of.

### CODE REVIEW

File: council/runners/parallel.py [hunk 1]
```diff
@@ -16,13 +15,14 @@ def __init__(self, *runners: RunnerBase):
 
     def _run(
         self,
-        context: ChainContext,
-        budget: Budget,
+        context: RunnerContext,
         executor: RunnerExecutor,
     ) -> None:
-        fs = [executor.submit(lambda: runner.run(context, budget, executor)) for runner in self.runners]
+        contexts = [(runner, context.fork()) for runner in self.runners]
```
Comment (aflament) [inline council/runners/parallel.py:21]: Add a comment to explain the threading issue we faced

File: docs/source/reference/runners/parallel_for.rst [hunk 2]
```diff
@@ -48,7 +48,7 @@ This example builds on the previous one and shows how to consume the iteration i
             print(message)
             return self.build_success_message(message=message)
 
-    chain = Chain(name="name", description="parallel for", runners=[ParallelFor(generator, MySkill(), parallelism=1)])
+    chain = Chain(name="name", description="parallel for", runners=[ParallelFor(generator, MySkill(), parallelism=5)])
```
Comment (aflament) [inline docs/source/reference/runners/parallel_for.rst:51]: Should we introduce an env var with a default value equal to 5 `COUNCIL_DEFAULT_PARALLELISM` ?

**Review by aflament — APPROVED** [review]
🚀

File: docs/source/reference/runners/parallel_for.rst (hunk 2 above)
Comment (gkoch78) [reply docs/source/reference/runners/parallel_for.rst:51]: that parameter is directly accessible when building a Chain. I'm not sure an EnvVar would add much value here

**Review by aflament — APPROVED** [review]

**Pull request merged**